x-pack/legacy/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/categorization_view/examples_valid_callout.tsx

```diff
@@ -60,45 +57,24 @@ export const ExamplesValidCallout: FC<Props> = ({
 
   return (
     <EuiCallOut color={color} title={title}>
-      {percentageText}
+      {validationChecks.map((v, i) => (
+        <div key={i}>{v.message}</div>
+      ))}
       <EuiSpacer size="s" />
       {analyzerUsed}
     </EuiCallOut>
   );
 };
 
-const PercentageText: FC<{ examplesValid: number; sampleSize: number }> = ({
-  examplesValid,
-  sampleSize,
-}) => (
-  <div>
-    <FormattedMessage
-      id="xpack.ml.newJob.wizard.pickFieldsStep.categorizationFieldPercentage"
-      defaultMessage="{number} field {number, plural, zero {value} one {value} other {values}} analyzed, {percentage}% contain valid tokens."
-      values={{
-        number: sampleSize,
-        percentage: Math.floor(examplesValid * 100),
-      }}
-    />
-  </div>
-);
-
-const AnalyzerUsed: FC<{ categorizationAnalyzer: CategorizationAnalyzerType }> = ({
+const AnalyzerUsed: FC<{ categorizationAnalyzer: CategorizationAnalyzer }> = ({
   categorizationAnalyzer,
 }) => {
   let analyzer = '';
-  if (typeof categorizationAnalyzer === null) {
-    return null;
-  }
 
-  if (typeof categorizationAnalyzer === 'string') {
-    analyzer = categorizationAnalyzer;
-  } else {
-    if (categorizationAnalyzer?.tokenizer !== undefined) {
-      analyzer = categorizationAnalyzer?.tokenizer!;
-    } else if (categorizationAnalyzer?.analyzer !== undefined) {
-      analyzer = categorizationAnalyzer?.analyzer!;
-    }
+  if (categorizationAnalyzer?.tokenizer !== undefined) {
+    analyzer = categorizationAnalyzer?.tokenizer!;
```

Comment:
Is the optional chaining (?) and non-null assertion operator (!) still needed here even after the `undefined` check? 🤔

Reply:
fixed in 062cdff86885d2156709b5f528c0a90c8d053358